lib/elixir/lib/module/types.ex

```diff
@@ -236,6 +236,9 @@ defmodule Module.Types do
       map_type?(left) and map_type?(right) and match?({:ok, _}, missing_field(left, right)) ->
         {:ok, atom} = missing_field(left, right)
 
+        # Drop the last trace which is the expression map.foo
+        traces = Enum.drop(traces, 1)
```

Comment:
Can we get the line from the trace though and print it when showing the `expr` below? Otherwise we show the location information only at the bottom and the trace information is very handy!

Reply:
Should we do this on all errors where we print the expressions or just this one?

Reply:
You can see other examples in `checker_test.exs`.

Reply:
I think all of them would look better with it but I defer the final judgement to you :)